Src/Support/Google.Apis.Auth/OAuth2/AwsExternalAccountCredential.AwsSignedSubjectToken.cs

```diff
@@ -0,0 +1,279 @@
+﻿/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Util;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    public partial class AwsExternalAccountCredential
+    {
+        /// <summary>
+        /// Partial implementation of AWS signed request, enough to support signing
+        /// of a, usually, AWS GetCallerIdentity request. The signed request is sent
+        /// to Google's subject token service as the subject token to be exchanged for
+        /// access tokens. Google STS triggers the request as specified by the signed
+        /// request to verify the callers identity.
+        /// </summary>
+        private sealed class AwsSignedSubjetToken
+        {
+            private const string StVerificationHttpMethod = "POST";
+            private const string BasicFormatIso8601 = "yyyyMMddTHHmmssZ";
```

Comment:
I'd put T and Z in quotes. I *think* they're okay as they are, strictly speaking, but it makes it clearer that they're literal values rather than placeholders.